### ISSUE TITLE
Moving some information from "Valid Usage" to field or function parameter descriptions

### ISSUE THREAD

**Issue by elfring** [KhronosGroup/Vulkan-Docs#39]
I skimmed the current application programming interface specification to some degree. Now I wonder about the usefulness of information which is provided in [sections like "Valid Usage"](https://www.khronos.org/registry/vulkan/specs/1.0/man/html/VkAllocationCallbacks.html#_valid_usage "Descriptions for the valid usage of items").

I find that several descriptions there should belong to the explanation of data structure members or function parameters directly.
* How do you think about any more fine-tuning around involved data type constraints?
* Would you like to reduce the presentation of duplicate information?

**Comment by oddhack** [comment]
Could you give some specific examples?

**Comment by elfring** [comment]
Are descriptions around [the API item "VkAllocationCallbacks"](https://www.khronos.org/registry/vulkan/specs/1.0/man/html/VkAllocationCallbacks.html#_valid_usage "Descriptions for the valid usage of items") interesting update candidates?

**Comment by oddhack** [comment]
I'm still not quite sure what you would like done here. I will say that the ref pages, like the one you linked to, are in some cases not entirely up to date with the API spec (mostly because the ref pages are largely the output of a single heroic person while the spec has been worked on by everyone in the Khronos working group in parallel). So it's possible that some of the API spec language just needs to be imported to the ref page to address your concern - maybe you could compare side by side with https://www.khronos.org/registry/vulkan/specs/1.0/xhtml/vkspec.html#VkAllocationCallbacks and see if the API spec description would address your concern, if the ref page description were updated to be more similar?

(N.b. we would have liked to find a way to pull most of the ref page content directly from the spec instead of maintaining it separately, but we haven't been able to do that yet, which is how inconsistencies can arise.)

**Comment by elfring** [comment]
* How much do you care for [the description of null pointer handling](https://en.wikipedia.org/wiki/Null_pointer "Constraints for null pointers") directly next to each identifier in the application programming interface for example?
* Would you like to add any links for [the applied software design patterns](https://en.wikipedia.org/wiki/Software_design_pattern "Description for the usage of design patterns") (or other background information)?

**Comment by elfring** [comment]
>(N.b. we would have liked to find a way to pull most of the ref page content directly from the spec instead of maintaining it separately, but we haven't been able to do that yet, which is how inconsistencies can arise.)

* How does such background information fit to your tagging of this issue as "wontfix"?
* Are you looking for further possibilities to extend [the used RELAX NG schema](https://github.com/KhronosGroup/Vulkan-Docs/blob/d204ac2f6991f75fab2ebc22136fd179d692f84f/src/spec/registry.rnc#L246 "Specification for API properties")?
* Will an element like "validity" result in interesting software management challenges?

**Comment by oddhack** [comment]
The ref pages will get more complete and up to date coverage as the ref page editor has time for it. We will extend the schema when required to support specific needs. We will deal with specific software management challenges as they arise.

**Comment by elfring** [comment]
Would you like to achieve that the reference specification can be completely updated (and not only Asciidoc include files and other headers) by the combination of source files like the following?
* [vk.xml](https://github.com/KhronosGroup/Vulkan-Docs/blob/5a4c5e5925c65c6e6677c1fb21571684b4b0a77b/src/spec/vk.xml "XML API description")
* [registry.rnc](https://github.com/KhronosGroup/Vulkan-Docs/blob/d204ac2f6991f75fab2ebc22136fd179d692f84f/src/spec/registry.rnc#L246 "RELAX NG compact schema for validating XML")
